WordPress/Classes/ViewRelated/NUX/SigninSelfHostedViewController.swift

```diff
@@ -4,9 +4,8 @@ import WordPressShared
 /// Provides a form and functionality to sign-in and add an existing self-hosted 
 /// site to the app.
 ///
-@objc class SigninSelfHostedViewController : NUXAbstractViewController, SigninKeyboardResponder
+@objc class SigninSelfHostedViewController : NUXAbstractViewController, SigninKeyboardResponder, SigninWPComSyncHandler
```

Comment:
I feel like the abstraction is breaking a bit. `SigninWPComSyncHandler` is defined as:

> A protocol and extension encapsulating syncing behavior common to WPCom signin controllers

And yet, we have a `SelfHosted` controller that conforms to this protocol.

This leads to inconsistencies like having to implement an empty `updateSafariCredentialsIfNeeded` because we don't want credentials saved for "self hosted" sites. Or do we? If we just logged in to a wp.com site through this controller, it'd be interesting to update the credentials.